src/algebraic_geometry/presheafed_space.lean

```diff
@@ -103,21 +104,31 @@ variables {C}
 instance {X Y : PresheafedSpace.{v} C} : has_coe (X ⟶ Y) (X.to_Top ⟶ Y.to_Top) :=
 { coe := λ α, α.f }
 
-@[simp] lemma id_f (X : PresheafedSpace.{v} C) : ((𝟙 X) : X ⟶ X).f = 𝟙 X.to_Top := rfl
+@[simp] lemma id_f (X : PresheafedSpace.{v} C) :
+  ((𝟙 X) : X ⟶ X).f = 𝟙 (X : Top.{v}) := rfl
+@[simp] lemma id_coe (X : PresheafedSpace.{v} C) :
+  (((𝟙 X) : X ⟶ X) : (X : Top.{v}) ⟶ X) = 𝟙 (X : Top.{v}) := rfl
 @[simp] lemma comp_f {X Y Z : PresheafedSpace.{v} C} (α : X ⟶ Y) (β : Y ⟶ Z) :
```

Comment:
There are four combinations that can occur here. Does it make sense to have `simp_f_to_coe`. In that way you could replace the four simp-lemmas that we have here with just `id_coe` and `comp_coe`?
If we decide that the coe is the canonical form, we should probably just simp the `.f`'s away.

Reply:
Yup, seems reasonable, and works. I did have to add
```
@[simp] lemma hom_mk_coe {X Y : PresheafedSpace.{v} C} (f) (c) :
  (({ f := f, c := c } : X ⟶ Y) : (X : Top.{v}) ⟶ (Y : Top.{v})) = f := rfl
```
which formerly would have worked merely by `dsimp` (is that iota reduction?).